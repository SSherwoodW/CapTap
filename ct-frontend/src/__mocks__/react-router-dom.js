import { vi } from "vitest";

export * from "react-router-dom";

export const useParams = vi.fn();
// .. add more mocks if needed
