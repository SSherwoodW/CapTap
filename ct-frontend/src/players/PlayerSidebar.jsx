import React from "react";
import { Link } from "react-router-dom";

function PlayerSidebar({ players, teamCode, playerId }) {

  return (
    <div className="Sidebar bg-slate-600 border-2 border-gray-500 rounded-sm my-4 mx-4 p-6 w-64">
          <h3 className="text-xl text-center py-2 mb-4 uppercase text-gray-300 bg-indigo-500 border-2 border-gray-500 rounded-sm">Parlayer</h3>
          <h6 className="mb-2 m-4 text-sm font-thin text-gray-200">Select a teammate</h6>
      <ul>
        {players.map((player) => (
          <Link to={`/parlayer/${teamCode}`} state={{player1Id: playerId, player2Id: player.id}} key={player.full_name}>
            <li className="py-1 px-0 text-gray-200 hover:bg-gray-800 rounded transition duration-300" >{player.full_name}</li>
          </Link>
        ))}
      </ul>
    </div>
  );
}

export default PlayerSidebar;