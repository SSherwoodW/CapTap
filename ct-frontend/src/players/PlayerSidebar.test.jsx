import React from 'react';
import { render } from '@testing-library/react';
import PlayerSidebar from './PlayerSidebar';
import { MemoryRouter } from 'react-router-dom';

describe('PlayerSidebar component', () => {
  const players = [
    { id: 1, full_name: 'Player 1' },
    { id: 2, full_name: 'Player 2' },
    { id: 3, full_name: 'Player 3' },
  ];
  const teamCode = 'team-code';
  const playerId = 1;

  it('should render without crashing', () => {
    render(
      <MemoryRouter>
        <PlayerSidebar players={players} teamCode={teamCode} playerId={playerId} />
      </MemoryRouter>
    );
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <PlayerSidebar players={players} teamCode={teamCode} playerId={playerId} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the correct number of players', () => {
    const { getAllByRole } = render(
      <MemoryRouter>
        <PlayerSidebar players={players} teamCode={teamCode} playerId={playerId} />
      </MemoryRouter>
    );
    const playerElements = getAllByRole('link');
    expect(playerElements).toHaveLength(players.length);
  });

  it('should render the player names', () => {
    const { getByText } = render(
      <MemoryRouter>
        <PlayerSidebar players={players} teamCode={teamCode} playerId={playerId} />
      </MemoryRouter>
    );
    players.forEach((player) => {
      const playerElement = getByText(player.full_name);
      expect(playerElement).toBeInTheDocument();
    });
  });
});
