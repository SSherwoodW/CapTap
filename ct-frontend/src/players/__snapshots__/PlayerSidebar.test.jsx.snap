// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerSidebar component > should match the snapshot 1`] = `
<div>
  <div
    class="Sidebar bg-slate-600 border-2 border-gray-500 rounded-sm my-4 mx-4 p-6 w-64"
  >
    <h3
      class="text-xl text-center py-2 mb-4 uppercase text-gray-300 bg-indigo-500 border-2 border-gray-500 rounded-sm"
    >
      Parlayer
    </h3>
    <h6
      class="mb-2 m-4 text-sm font-thin text-gray-200"
    >
      Select a teammate
    </h6>
    <ul>
      <a
        href="/parlayer/team-code"
      >
        <li
          class="py-1 px-0 text-gray-200 hover:bg-gray-800 rounded transition duration-300"
        >
          Player 1
        </li>
      </a>
      <a
        href="/parlayer/team-code"
      >
        <li
          class="py-1 px-0 text-gray-200 hover:bg-gray-800 rounded transition duration-300"
        >
          Player 2
        </li>
      </a>
      <a
        href="/parlayer/team-code"
      >
        <li
          class="py-1 px-0 text-gray-200 hover:bg-gray-800 rounded transition duration-300"
        >
          Player 3
        </li>
      </a>
    </ul>
  </div>
</div>
`;
